src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -70,7 +70,7 @@ internal sealed class State
                 SingleWriter = true,
             });
 
-            public void RemoveStream(MsQuicStream stream)
+            public void RemoveStream(MsQuicStream? stream)
```

Comment:
I was going back and forth on this one. When debugging it is very _nice_ to have the stream here so one can emit traces from single location. This is basically same for the AddStream.

Reply:
I'm trying to understand why we need this at all... It seems like we are deferring Dispose of the connection until all streams in use on it are completed. Is that correct? Why do we want to do this? Can we just get rid of this tracking completely?

Reply:
yes, that is the intend. We saw crashes without it but it can perhaps be done in different way. If so I would do it in separate PR. The reason why this was touches is that the HandleEventShutdownComplete does not have access to the Stream itself so it would pass NULL. 
I'm OK removing the parameter completely as it is not use for product or perhaps using it Debug builds only and for example dump the Stream info if we are on path to Assert.

Reply:
I agree, we should consider for a separate PR.

Mostly I want to make sure we are making an intentional decision about behavior here.

Reply:
When chatting with  @nibanks he sad we would not get HandleEventShutdownComplete with pending streams. So if we delay cleanup in Dispose and if we can relay on that we could perhaps remove to logic. It would be nice to have stress tests up at that time so we can verify the impact.